src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1068,6 +1076,91 @@
     "alter table reports add column corrective_change boolean"
     "alter table resource_events add column corrective_change boolean"))
 
+(defn migrate-through-app
+  "stream data from table1 through the application and into table2, applying
+   munge-fn to each row"
+  [table1 table2 column-list batchsize munge-fn] 
+  (let [columns (string/join "," column-list)]
+    (jdbc/query-with-resultset
+      [(format "select %s from %s" columns (name table1))]
+      (fn [rs]
+        (->> (sql/result-set-seq rs)
+             (map munge-fn)
+             (partition-all batchsize)
+             (map #(insert-records* (name table2) %))
+             dorun)))))
+
+(defn resource-params-cache-parameters-to-jsonb
+  []
+  (jdbc/do-commands
+    (sql/create-table-ddl :resource_params_cache_transform
+                          ["resource" "bytea NOT NULL"]
+                          ["parameters" "jsonb"]))
+
+  (migrate-through-app
+    :resource_params_cache
+    :resource_params_cache_transform
+    ["encode(resource::bytea, 'hex') as resource" "parameters"]
+    500
```

Comment:
@senior I just made this number up, figuring it corresponded to the size of a large factset. We could probably make it larger to reduce the number of insert statements required, at the cost of more memory consumption.